plugins/destination/snowflake/client/read.go

```diff
@@ -24,67 +24,112 @@ func (c *Client) reverseTransform(f arrow.Field, bldr array.Builder, val any) er
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse int16: %w", err)
 		}
 		b.Append(int16(u))
 	case *array.Int32Builder:
 		u, err := strconv.ParseInt(val.(string), 10, 32)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse int32: %w", err)
 		}
 		b.Append(int32(u))
 	case *array.Int64Builder:
 		u, err := strconv.ParseInt(val.(string), 10, 64)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse int64: %w", err)
 		}
 		b.Append(u)
 	case *array.Uint8Builder:
 		u, err := strconv.ParseUint(val.(string), 10, 8)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse uint8: %w", err)
 		}
 		b.Append(uint8(u))
 	case *array.Uint16Builder:
 		u, err := strconv.ParseUint(val.(string), 10, 16)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse uint16: %w", err)
 		}
 		b.Append(uint16(u))
 	case *array.Uint32Builder:
 		u, err := strconv.ParseUint(val.(string), 10, 32)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse uint32: %w", err)
 		}
 		b.Append(uint32(u))
 	case *array.Uint64Builder:
 		u, err := strconv.ParseUint(val.(string), 10, 64)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to parse uint64: %w", err)
 		}
 		b.Append(u)
 	case *array.Float32Builder:
-		b.Append(val.(float32))
+		if floatVal, ok := val.(float64); ok {
+			b.Append(float32(floatVal))
+		} else {
+			floatVal, err := strconv.ParseFloat(val.(string), 32)
+			if err != nil {
+				return fmt.Errorf("failed to parse float32: %w", err)
+			}
+			b.Append(float32(floatVal))
+		}
 	case *array.Float64Builder:
-		b.Append(val.(float64))
+		if floatVal, ok := val.(float64); ok {
+			b.Append(floatVal)
+		} else {
+			floatVal, err := strconv.ParseFloat(val.(string), 64)
+			if err != nil {
+				return fmt.Errorf("failed to parse float64: %w", err)
+			}
+			b.Append(floatVal)
+		}
 	case *array.StringBuilder:
 		b.Append(val.(string))
 	case *array.LargeStringBuilder:
 		b.Append(val.(string))
 	case *array.BinaryBuilder:
 		b.Append(val.([]uint8))
 	case *array.TimestampBuilder:
-		b.Append(arrow.Timestamp(val.(time.Time).UnixMicro()))
+		var timeVal time.Time
+		if t, ok := val.(time.Time); ok {
+			timeVal = t
+		} else {
+			parsedTime, err := time.Parse("2006-01-02 15:04:05.999999999", val.(string))
+			if err != nil {
+				return fmt.Errorf("failed to parse timestamp: %w", err)
+			}
+			timeVal = parsedTime
+		}
+		switch b.Type().(*arrow.TimestampType).Unit {
+		case arrow.Second:
+			b.Append(arrow.Timestamp(timeVal.UTC().Unix()))
+		case arrow.Millisecond:
+			b.Append(arrow.Timestamp(timeVal.UTC().UnixMilli()))
+		case arrow.Microsecond:
+			b.Append(arrow.Timestamp(timeVal.UTC().UnixMicro()))
+		case arrow.Nanosecond:
+			b.Append(arrow.Timestamp(timeVal.UTC().UnixNano()))
+		default:
+			return fmt.Errorf("unsupported timestamp unit %s", f.Type.(*arrow.TimestampType).Unit)
+		}
 	case array.ListLikeBuilder:
```

Comment:
just use from string

Reply:
basically we need to add to arrow `AppendTime` but for now not to block it we can just use this.